lib/nostrum/struct/channel.ex

```diff
@@ -1,27 +1,124 @@
 defmodule Nostrum.Struct.Channel do
-  @moduledoc ~S"""
+  @moduledoc """
+  Struct representing a channel.
 
-  A `Nostrum.Struct.Channel` represents a Guild or DM channel.
+  There are currently eleven channel types within discord, ranging from direct messages between two users to server announcement channels pushing a message to thousands. This module also contains several helper functions for handling channels within your applcation.
```

Comment:
I think this could be cleaned up a bit. Maybe change the top line to "Struct and helper functions for working with channels."? We could then remove the second line completely. If you really want to include the bit about the different channel types then we could move it elsewhere. Up to you!

Reply:
I also think we could consider this a `context` module as it'd be called in Phoenix. Just something to think about 🤷